frontend/src/InputForm.js

```diff
@@ -7,37 +7,39 @@ import BasicInfo from './BasicInfo';
 class InputForm extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {
-      courses: [{
-        name: '',
-        courseID: '',
-        subject: '',
-        credits: '',
-        isRequired: false,
-        rank: '',
-        lectureSections: [{
-          professor: '',
-          startTime: '',
-          endTime: '',
-          days: [],
+    this.state =
+      JSON.parse(window.localStorage.getItem('inputFormState')) ||
+      {
+        courses: [{
+          name: '',
+          courseID: '',
+          subject: '',
+          credits: '',
+          isRequired: false,
+          rank: '',
+          lectureSections: [{
+            professor: '',
+            startTime: '',
+            endTime: '',
+            days: [],
+          }],
+          labSections: [],
         }],
-        labSections: [],
-      }],
-      criterion: {
-        timePreferences: [],
-        preferredSubject: '',
-      },
-      basicInfo: {
-        credits: {
-          minCredits: 12,
-          maxCredits: 18,
+        criterion: {
+          timePreferences: [],
+          preferredSubject: '',
         },
-        termDates: {
-          startDate: '',
-          endDate: '',
+        basicInfo: {
+          credits: {
+            minCredits: 12,
+            maxCredits: 18,
```

Comment:
Why are the `minCredits/maxCredits` set? Do they get overwritten?

Reply:
These have been set as defaults - this isn't a change made in this pull request. If we do want to change it (which makes sense given that the user's min/max credits could definitely be different from the default) it would be worth opening a separate PR.

Reply:
Ah ok, I forgot it was defaulted before. Up to you, its ok as is.